### PR TITLE
numpy 2.0.0rc1

### PR DESCRIPTION
**Pull request by h-vetinari**
Reactivate `dev` branch for #311

### CODE REVIEW

**Comment by mattip** [comment]
It seems the windows entry-point problem is generic and not connected to this PR? When I reproduce locally, the entry-point is trying to use a non-existant python:
```
>conda build recipe -m .ci_support\win_64_blas_implmklnumpy1.26python3.12.____cpythonpython_implcpython.yaml
<wait for build to fail, look at the test environment path>
>conda activate d:\miniconda\envs\builder\conda-bld\numpy_1710845979455\_test_env
>numpy-config
Fatal error in launcher: Unable to create process using '"d:\miniconda\envs\builder\conda-bld\numpy_1710845979455\_h_env\python.exe"  "d:\miniconda\envs\builder\conda-bld\numpy_1710845979455\_test_env\Scripts\numpy-config.exe" ': The system cannot find the file specified.
```

The fatal error is correct, there is no `_h_env`. If I copy `_test_env` to `_h_env`, then the entry-point works.

**Comment by rgommers** [comment]
The entrypoint needs to be added to the recipe:

https://github.com/conda-forge/numpy-feedstock/blob/b60684fab0eae8c1d9f51ba30adcd03c8691bf87/recipe/meta.yaml#L27-L28

**Comment by rgommers** [comment]
It looks like that is the conda-forge recipe equivalent of this `pyproject.toml` content:
```toml
[project.scripts]
f2py = 'numpy.f2py.f2py2e:main'
numpy-config = 'numpy._configtool:main'
```

**Comment by jakirkham** [comment]
Possibly seeing issue ( https://github.com/numpy/numpy/issues/25940 ) in macOS `x86_64` builds [on CI]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=899274&view=logs&j=9c5ef928-2cd6-52e5-dbe6-9d173a7d951b&t=20c71c51-4b27-578b-485d-06ade2de1d00&l=1914 ):

```cpp
[474/517] Compiling C++ object numpy/fft/_pocketfft_umath.cpython-310-darwin.so.p/_pocketfft_umath.cpp.o
FAILED: numpy/fft/_pocketfft_umath.cpython-310-darwin.so.p/_pocketfft_umath.cpp.o 
x86_64-apple-darwin13.4.0-clang++ -Inumpy/fft/_pocketfft_umath.cpython-310-darwin.so.p -Inumpy/fft -I../numpy/fft -Inumpy/_core -I../numpy/_core -Inumpy/_core/include -I../numpy/_core/include -I../numpy/_core/src/common -I$PREFIX/include/python3.10 -I$SRC_DIR/builddir/meson_cpu -fvisibility=hidden -fvisibility-inlines-hidden -fcolor-diagnostics -DNDEBUG -Wall -Winvalid-pch -std=c++17 -O3 -ftrapping-math -DNPY_HAVE_CLANG_FPSTRICT -msse -msse2 -msse3 -mssse3 -DNPY_HAVE_SSE2 -DNPY_HAVE_SSE -DNPY_HAVE_SSE3 -DNPY_HAVE_SSSE3 -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fstack-protector-strong -O2 -pipe -stdlib=libc++ -fvisibility-inlines-hidden -fmessage-length=0 -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/numpy-2.0.0b1 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix -D_FORTIFY_SOURCE=2 -isystem $PREFIX/include -mmacosx-version-min=10.9 -MD -MQ numpy/fft/_pocketfft_umath.cpython-310-darwin.so.p/_pocketfft_umath.cpp.o -MF numpy/fft/_pocketfft_umath.cpython-310-darwin.so.p/_pocketfft_umath.cpp.o.d -o numpy/fft/_pocketfft_umath.cpython-310-darwin.so.p/_pocketfft_umath.cpp.o -c ../numpy/fft/_pocketfft_umath.cpp
In file included from ../numpy/fft/_pocketfft_umath.cpp:24:
../numpy/fft/pocketfft/pocketfft_hdronly.h:163:15: error: no member named 'aligned_alloc' in the global namespace; did you mean simply 'aligned_alloc'?
  void *ptr = ::aligned_alloc(align,(size+align-1)&(~(align-1)));
              ^~~~~~~~~~~~~~~
              aligned_alloc
../numpy/fft/pocketfft/pocketfft_hdronly.h:160:14: note: 'aligned_alloc' declared here
inline void *aligned_alloc(size_t align, size_t size)
```

**Comment by h-vetinari** [comment]
Yeah, the macOS thing is the last thing we need to fix from the POV of conda-forge (the entrypoint just needed to be specified, and the QEMU-related aarch failures we can skip). We could go to a minimum of 10.13, which is likely enough to make things work, but still the upstream pocketfft code is _intended_ to allow older macOS, and fails.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{% set version = "1.23.1" %}
+{% set version = "2.0.0b1" %}
```
Comment (h-vetinari) [inline recipe/meta.yaml:1]:
```suggestion
{% set version = "2.0.0rc1" %}
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -9,23 +9,27 @@ package:
 
 source:
   - url: https://github.com/numpy/numpy/releases/download/v{{ version }}/numpy-{{ version }}.tar.gz
-    sha256: d748ef349bfef2e1194b59da37ed5a29c19ea8d7e6342019921ba2ba4fd8b624
+    sha256: e0bb33a37d0d0b9a19cd41a093877f830e06bd4d989341b9792896cf08e629f7
```
Comment (h-vetinari) [inline recipe/meta.yaml:12]:
```suggestion
    sha256: f0e169ec6cbc1b8e5f6a235845a80961f76f88352082213a1728a0967a761ad2
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -52,88 +62,92 @@ requirements:
...
 # there are some tests that cannot really work in emulation, see e.g. numpy/numpy#20445
 {% set tests_to_skip = tests_to_skip + " or Test_ARM_Features" %}                   # [build_platform != target_platform]
+{% set tests_to_skip = tests_to_skip + " or Test_POWER_Features" %}                 # [build_platform != target_platform]
 # test_new_policy reruns part of test suite; including a CPU feature test that fails in emulation
 {% set tests_to_skip = tests_to_skip + " or test_new_policy" %}                     # [build_platform != target_platform]
-# flaky refcount-based test; already skipped upstream for win+py39
-{% set tests_to_skip = tests_to_skip + " or test_partial_iteration_cleanup" %}      # [osx]
-
+# emulation problems (apparently) on aarch
+{% set tests_to_skip = tests_to_skip + " or test_basic_property[float32" %}         # [aarch64]
```
Comment (h-vetinari) [inline recipe/meta.yaml:71]:
```suggestion
{% set tests_to_skip = tests_to_skip + " or (test_basic_property and float32)" %}   # [aarch64]
```

**Comment by h-vetinari** [comment]
@isuruf
I'm only on a phone ATM, could you perhaps lend a hand to rebase out some debug commits here?

The following should be removed:
- https://github.com/conda-forge/numpy-feedstock/pull/312/commits/e5ff87e97a08ccd7e7ded060066dcfdc7c60780d
- https://github.com/conda-forge/numpy-feedstock/pull/312/commits/944294e24e3c0b21f77ff07b9c102f2ff02e8523
- https://github.com/conda-forge/numpy-feedstock/pull/312/commits/f6e8e56349acdc8891e0cccfb60063b3ca074d9f

Plus a fresh rerender afterwards to be sure... 

For context, I'd like to merge 2.0.0rc1 into the `numpy_dev` label soon so that we can start testing against it.

**Comment by isuruf** [comment]
@conda-forge-admin, rerender

**Comment by h-vetinari** [comment]
Thank you! I guess it's not a big deal to keep https://github.com/conda-forge/numpy-feedstock/pull/312/commits/dab58400cb3340a818530dfed867c8363a87559a for now. :)

**Comment by jakirkham** [comment]
Thanks all! 🙏

**Pull request merged**